jspoon/src/main/java/pl/droidsonroids/jspoon/ArrayUtils.java

```diff
@@ -0,0 +1,168 @@
+package pl.droidsonroids.jspoon;
+
+class ArrayUtils {
+
+    /**
+     * Converts a given array of wrappers to the same type of primitive array. Null values are
+     * represented as initial value for that primitive type. If the wrapper array is null, will
+     * return null.
+     * @param array wrapper array
+     * @param primitiveType type of the returned primitive array
+     * @throws IllegalArgumentException if the given primitive type is not a class of a primitive
+     * @throws ClassCastException if primitive type does not represent items of the wrapper array
+     */
+    static Object toPrimitive(final Object[] array, Class<?> primitiveType) {
+        if (array == null) {
+            return null;
+        }
+        if (!primitiveType.isPrimitive()) {
+            throw new IllegalArgumentException(primitiveType + " is not a primitive type");
+        }
+        Class<?> pt = primitiveType;
+        if(Boolean.TYPE.equals(pt)) {
+            return toPrimitive((Boolean[]) array);
+        }
+        if(Integer.TYPE.equals(pt)) {
+            return toPrimitive((Integer[]) array);
+        }
+        if(Long.TYPE.equals(pt)) {
+            return toPrimitive((Long[]) array);
+        }
+        if(Short.TYPE.equals(pt)) {
+            return toPrimitive((Short[]) array);
+        }
+        if(Byte.TYPE.equals(pt)) {
+            return toPrimitive((Byte[]) array);
+        }
+        if(Character.TYPE.equals(pt)) {
+            return toPrimitive((Character[]) array);
+        }
+        if(Double.TYPE.equals(pt)) {
+            return toPrimitive((Double[]) array);
+        }
+        if(Float.TYPE.equals(pt)) {
+            return toPrimitive((Float[]) array);
+        }
+        return array;
+    }
+
+    static char[] toPrimitive(Character[] array) {
+        if (array == null) {
+            return null;
+        }
+        final char[] result = new char[array.length];
+        for (int i = 0; i < array.length; i++) {
+            if (array[i] == null){
+                result[i] = (new char[1])[0];
```

Comment:
Why not just '\u0000` instead of `(new char[1])[0]`?

Reply:
Since performance difference is practically non-existent, it was done like this mainly for readability. If someone not familiar with JDK's default values for primitives would look at this code seeing `\u0000` could look like a value defined by a developer and this way it's very clear to be JDK's initial value even if you don't know what it is. Also this way the implementations for all primitive arrays follow the same pattern.